File: Wyvern.Front/src/app/personagens/models/personagem.ts
export interface Personagem {
    personagemId?: number;
    nome: string;
    descricao?: string;
    campanhaId: number;
    tipoId: number; // 1 = Jogador, 2 = NPC, 3 = Monstro
    criadoPorId?: number;
    criadoEm?: string;
    ativo?: boolean;
    atributo?: any;
    personagemPlayer?: any;
    personagemCombate?: any;
    personagemDetalhes?: any;
    personagemDinheiro?: any;
    personagemItens?: PersonagemItem[];
    personagemMagias?: PersonagemMagia[];
    personagemAtaques?: PersonagemAtaque[];
    personagemNpc?: any;
    personagemAcoesPadrao?: any[];
    personagemAcoesBonus?: any[];
    personagemReacoes?: any[];
    personagemAcoesLendarias?: any[];
    personagemTracosEspeciais?: any[];
}

export interface PersonagemItem {
    personagemItemId: number;
    personagemId: number;
    nome: string;
    tipoItem: string;
    raridade: string;
    descricao?: string;
}

export interface PersonagemMagia {
    personagemMagiaId?: number;
    personagemId: number;
    nome: string;
    nivel: number;
    escola?: string;
    verbal?: boolean;
    somatico?: boolean;
    material?: boolean;
    descricao?: string;
}

export interface PersonagemAtaque {
    personagemAtaqueId?: number;
    nome: string;
    alcance?: string;
    bonusAcerto: number;
    dano?: string;
    tipoDano?: string;
    propriedades?: string;
}

File: Wyvern/Wyvern.Domain/Entities/Personagem.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Wyvern.Domain.Entities
{
    public class Personagem
    {
        [Key]
        public int PersonagemId { get; set; }
        [Required]
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public int CampanhaId { get; set; }
        public Campanha Campanha { get; set; }
        public int TipoId { get; set; } // enum se vai ser jogador,npc ou monstro
        public TipoPersonagem Tipo { get; set; }
        public int CriadoPorId { get; set; }
        public Usuario CriadoPor { get; set; }
        public DateTime CriadoEm { get; set; }
        public Atributo Atributo { get; set; }
        public bool Ativo { get; set; } = true;
        public List<PersonagemItem>? PersonagemItens { get; set; }
        public List<PersonagemMagia>? PersonagemMagias { get; set; }
        public PersonagemPlayer? PersonagemPlayer { get; set; }
        public PersonagemCombate? PersonagemCombate { get; set; } 
        public List<PersonagemPericia>? PersonagemPericias { get; set; }
        public PersonagemDetalhes? PersonagemDetalhes { get; set; }
        public PersonagemConjuracao? PersonagemConjuracao { get; set; }
        public PersonagemDinheiro? PersonagemDinheiro { get; set; }
        public ICollection<PersonagemAtaque>? PersonagemAtaques { get; set; }

        public PersonagemNpc? PersonagemNpc { get; set; }
        public ICollection<PersonagemAcaoPadrao>? PersonagemAcoesPadrao { get; set; }
        public ICollection<PersonagemAcaoBonus>? PersonagemAcoesBonus { get; set; }
        public ICollection<PersonagemReacao>? PersonagemReacoes { get; set; }
        public ICollection<PersonagemAcaoLendaria>? PersonagemAcoesLendarias { get; set; }
        public ICollection<PersonagemTracoEspecial>? PersonagemTracosEspeciais { get; set; }
    }
}
